homeworks/23/chat-system/src/main/java/org/example/Client.java

```diff
@@ -0,0 +1,65 @@
+package org.example;
+
+import java.io.*;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.util.Scanner;
+
+public class Client {
+    public static void main(String[] args) {
+        var scanner = new Scanner(System.in);
+        String command = "";
+
+        while (!command.equals("/quit")) {
+            System.out.print("\033[35;3mEnter command\033[0m: ");
+            // System.out.print("Enter command: ");  // Commented lines do not apply color to the printed text (some terminals do not support coloring)
+            command = scanner.nextLine();
+            var input = command.split(" +");
+            if (input.length != 2) {
+                continue;
+            }
+            if (input[0].equals("/connect")) {
+                var ipAndPort = input[1].split(":");
+                if (ipAndPort.length != 2) {
+                    continue;
+                }
+                connectToServer(ipAndPort[0], Integer.parseInt(ipAndPort[1]));
+            }
+        }
+    }
+
+    private static void connectToServer(String hostname, int port) {
+        try (var socket = new Socket(hostname, port)) {
+            System.out.println("Connected to \033[34;3m" + socket.getInetAddress().getHostAddress() + "\033[0;3m:\033[33;3m" + socket.getPort() + "\033[0m");
+            // System.out.println("Connected to " + socket.getInetAddress().getHostAddress() + ":" + socket.getPort());
+
+            OutputStream out = socket.getOutputStream();
+            var output = new DataOutputStream(out);
+            InputStream in = socket.getInputStream();
+            var input = new BufferedReader(new InputStreamReader(in));
+
+            var userInput = new Scanner(System.in);
+            String command = "";
+
+            while (!command.equals("/quit")) {
```

Comment:
The way that you receive messages means that they are received only when pressing enter. You should use a new thread for reading.